Tests/Integration/MSFT_xMsiPackage.Integration.Tests.ps1

```diff
@@ -0,0 +1,338 @@
+                        -Ensure 'Absent' `
+                        -Path $script:msiLocation `
+                        -ProductId $script:packageId
+
+                    $testTargetResourceResult | Should Be $true
+                }
+
+                It 'Should return correct value when package is present' {
+                    Set-TargetResource -Ensure 'Present' -Path $script:msiLocation -ProductId $script:packageId
+
+                    Test-PackageInstalledById -ProductId $script:packageId | Should Be $true
+
+                    $testTargetResourceResult = Test-TargetResource `
+                            -Ensure 'Present' `
+                            -Path $script:msiLocation `
+                            -ProductId $script:packageId `
+
+                    $testTargetResourceResult | Should Be $true
+
+                    $testTargetResourceResult = Test-TargetResource `
+                        -Ensure 'Absent' `
+                        -Path $script:msiLocation `
+                        -ProductId $script:packageId `
+
+                    $testTargetResourceResult | Should Be $false
+                }
+            }
+
+            Context 'Set-TargetResource' {
+                It 'Should correctly install and remove a .msi package' {
+                    Set-TargetResource -Ensure 'Present' -Path $script:msiLocation -ProductId $script:packageId
+
+                    Test-PackageInstalledById -ProductId $script:packageId | Should Be $true
+
+                    $getTargetResourceResult = Get-TargetResource -Path $script:msiLocation -ProductId $script:packageId
+
+                    $getTargetResourceResult.Version | Should Be '1.2.3.4'
+                    $getTargetResourceResult.InstalledOn | Should Be ('{0:d}' -f [DateTime]::Now.Date)
+                    $getTargetResourceResult.ProductId | Should Be $script:packageId
+
+                    [Math]::Round($getTargetResourceResult.Size, 2) | Should Be 0.03
+
+                    Set-TargetResource -Ensure 'Absent' -Path $script:msiLocation -ProductId $script:packageId
+
+                    Test-PackageInstalledById -ProductId $script:packageId | Should Be $false
+                }
+
+                It 'Should throw with incorrect product id' {
+                    $wrongPackageId = '{deadbeef-80c6-41e6-a1b9-8bdb8a050272}'
+
+                    { Set-TargetResource -Ensure 'Present' -Path $script:msiLocation -ProductId $wrongPackageId } | Should Throw
+                }
+
+                It 'Should correctly install and remove a package from a HTTP URL' {
+                    $baseUrl = 'http://localhost:1242/'
+                    $msiUrl = "$baseUrl" + 'package.msi'
+
+                    $fileServerStarted = $null
+
+                    try
+                    {
+                        $fileServerStarted = New-Object -TypeName 'System.Threading.EventWaitHandle' -ArgumentList @($false, [System.Threading.EventResetMode]::ManualReset,
```

Comment:
Suggestion: A little refactoring may improve test readability and also make new tests easier to write.

Creating the EventWaitHandle can be done by the server and then the handle can be returned by Start-Server along with the job object.

Disposing the handle, Stop-Job, and Remove-Job could be done with a new function Stop-Server.

Also, clearing the server's log should be done by the server (at the beginning of Start-Server).

Consider this comment as optional, you can leave the tests as they are now